### PR TITLE
Port lmnn meta example

### PR DESCRIPTION
**Pull request by LiuYuHui**

### CODE REVIEW

File: examples/meta/src/multiclass/lmnn.sg.in [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#![create_features]
+Features features_train = create_features(f_feats_train)
+Features features_test = create_features(f_feats_test)
+Labels labels_train = create_labels(f_labels_train)
+#![create_features]
+
+#![choose_distance]
+int k = 3
+LMNN lmnn(features_train, labels_train, k)
```
Comment (karlnapf) [inline examples/meta/src/multiclass/lmnn.sg.in:13]: is there no factory for this yet?

File: examples/meta/src/multiclass/lmnn.sg.in [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+lmnn.train()
+Distance lmnn_distance = lmnn.get_distance()
```
Comment (karlnapf) [inline examples/meta/src/multiclass/lmnn.sg.in:15]: can you already use `get` here?

**Review by karlnapf — COMMENTED** [review]
LGTM apart from the comments I hade

File: examples/meta/src/multiclass/lmnn.sg.in (hunk 1 above)
Comment (LiuYuHui) [reply examples/meta/src/multiclass/lmnn.sg.in:13]: it seems like that there is not factory for `LMNN`.
https://github.com/shogun-toolbox/shogun/blob/c964c9d1aab4bc1cf9133baf14d3bd5b96ba42de/src/shogun/metric/LMNN.h#L32

Comment (karlnapf) [reply examples/meta/src/multiclass/lmnn.sg.in:13]: ok that is fine for now

File: examples/meta/src/multiclass/lmnn.sg.in [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#![choose_distance]
+
+#![create_instance]
+Machine knn = create_machine("KNN", k=k,distance=lmnn_distance,labels=labels_train)
+#![create_instance]
+
+#![train_and_apply]
+knn.train()
+MulticlassLabels labels_predict = knn.apply(features_test)
```
Comment (gf712) [inline examples/meta/src/multiclass/lmnn.sg.in]:
```suggestion
Labels labels_predict = knn.apply(features_test)
```

File: examples/meta/src/multiclass/lmnn.sg.in (hunk 2 above)
Comment (gf712) [reply examples/meta/src/multiclass/lmnn.sg.in:15]: it seems like the Distance object is generated by `get_distance` and there it is not registered with `SG_ADD`

Comment (karlnapf) [reply examples/meta/src/multiclass/lmnn.sg.in:15]: mmmh ... ok maybe we can deal with this later....an issue maybe?

Comment (gf712) [reply examples/meta/src/multiclass/lmnn.sg.in:15]: agreed, it's better to have these integration tests merged in first and then later refactor to make sure we don't break stuff..

**Comment by gf712** [comment]
Thanks!

**Pull request merged**